src/inspector_agent.cc

```diff
@@ -205,8 +202,33 @@ std::string GenerateID() {
   return uuid;
 }
 
-}  // namespace
+std::string StringViewToUtf8(const StringView& view) {
+  if (view.is8Bit())
+    return std::string(reinterpret_cast<const char*>(view.characters8()),
+        view.length());
+
+  const uint16_t* source = view.characters16();
+  std::string result(view.length() * sizeof(*source), '\0');
+  UnicodeString utf16(reinterpret_cast<const UChar*>(source), view.length());
+  // ICU components for std::string compatibility are not enabled in build...
+  bool done = false;
+  while (!done) {
+    CheckedArrayByteSink sink(&result[0], result.length());
+    utf16.toUTF8(sink);
+    result.resize(sink.NumberOfBytesAppended());
+    done = !sink.Overflowed();
```

Comment:
Correct me if I'm wrong but CheckedArrayByteSink is instantiated every time with the same arguments in this loop so logically `sink.Overflowed()` is either always true or always false.  That means this loop either isn't (i.e., it's not a loop because it exits after the first iteration) or it's an infinite loop.